README.md

```diff
@@ -1,2 +1,478 @@
+- Type
+  - There can be many different types of Sources.  Typically this is
+  a cloud services provider.  Creating a new Type requires subclassing the
+  `Source` model.  See [Technical details](#technical-details).
+- Max Proxies
+  - Do not request or provision more than this many proxies from this Source.
+- Reliability Score
+  - It's possible, however unlikely, that two different sources can provide the
+  same IP address + port.  If that were to happen, the Source with the higher
+  Reliability Score will take ownership of that proxy.
+- Other
+  - Any other configuration parameters are specific to the Type of Source.
+
+#### Digital Ocean
+- proxy_port
+  - What port is exposed for proxy services on the droplet image.
+- client_id
+  - get at https://cloud.digitalocean.com/api_access
+- api_key
+  - get at https://cloud.digitalocean.com/api_access
+- image_name
+  - It's assumed that you have provisioned a server on your own and saved
+  an image with this name.  New proxies get created using this image.
+- flavor_name
+  - We recommend the '512MB' flavor
+- region_name
+  - This could be `New York 3`, `San Francisco 1`, etc.
+
+If any of the information entered above information is inaccurate
+then warning messages should
+show up on the page for that site shortly after the first proxy is
+requested from that source.
+
+## Settings
+Environment variables stored in redis.  These are populated on deploy from
+config/default_settings.yml.
+- success_ratio_threshold
+  - Each proxy is periodically evaluated for its ability to be used on each
+  site.  If the success_count/(success_count+failure_count) is less than this
+  number, then the proxy may be decommissioned.
+- default_retry_interval
+  - How long to tell clients to wait for proxies to be created when we have
+  none.
+- server_retry_timeout
+  - How long to wait for newly provisioned proxies to be ready before giving up.
+  - These proxies aren't lost forever.  They can be reclaimed by the system
+  later.
+- proxy_age_timeout_seconds
+  - If proxies haven't been requested for a site in this many seconds then
+  the site forgets that it had ever used the proxies that are presently
+  allocated to it.
+  - If there are no more sites presently reserving these proxies then the
+  proxies are decommissioned.
+- failure_threshold
+  - If this many failures happen in between runs of the periodic performance
+  analyzer then evaluate this proxy's performance early.
+  - If the number of successes is too low then the proxy will be removed from
+  the site's proxy pool, and possibly decommissioned.
+
+## API Keys
+API keys can be created or destroyed.  Use the key in the api_key parameter
+with the API.
+
+# Installation and deploy instructions
+
+Before you deploy to production you'll need to create a development environment.
+Your development environment will only require some of the same steps as
+deploying to production.  Install the core apt-get dependencies and rvm.
```

Comment:
OK.  I'll change the wording, but I do want to mention that there may be other dependencies not resolved by the following code blob.